app/test/package/package_publisher_test.dart

```diff
@@ -77,6 +81,42 @@ void main() {
     });
   });
 
+  group('Upload with a publisher', () {
+    testWithServices('not an admin', () async {
+      final client = createPubApiClient(authToken: hansUser.userId);
+      await client.setPackagePublisher(
+        hydrogen.package.name,
+        PackagePublisherInfo(publisherId: 'example.com'),
+      );
+      await dbService.commit(inserts: [
+        publisherMember(hansUser.userId, 'non-admin'),
```

Comment:
Are you testing a role other than admin here?

Reply:
Exactly. With the current API it is not really an approved value, so I couldn't set it that way, just "hacking" the DB for a non-admin value instead, which should be future-proof in the test if we ever introduce more roles.

Reply:
@isoos, it might be good to document that you're testing what happens if the datastore is in an invalid state. or not this somehow, that this is not currently possible.